### DISCUSSION TITLE
question about filtering

### DISCUSSION THREAD

**Discussion by farzadha2** [postalserver/postal#3115] [Help with using Postal]
Hi
I was wondering if its possible to filter by opened as it seems that it does not have that option
I also tried postal console to find class object but couldn't seem to find the correct one 

```
["WorkerRole",
 "WebhookRequest",
 "WebhookEvent",
 "Webhook",
 "UserInvite",
 "User",
 "TrackDomain",
 "Statistic",
 "SMTPEndpoint",
 "Server",
 "ScheduledTask",
 "Route",
 "QueuedMessage",
 "OrganizationUser",
 "OrganizationIPPool",
 "Organization",
 "IPPoolRule",
 "IPPool",
 "IPAddress",
 "HTTPEndpoint",
 "Domain",
 "Credential",
 "AddressEndpoint",
 "AdditionalRouteEndpoint",
 "ApplicationRecord",
 "Authie::SessionModel"]

```
not sure where it stores that part

**Comment by willpower232** [comment]
Unfortunately there isn't a way of filtering to opened, you would have to use `status: sent` and then manually spot the ones with the opened tag.

The messages are stored in sub-databases so they have their own code slightly outside of Rails. Have a look at the files under https://github.com/postalserver/postal/blob/main/lib/postal/message_db

**Reply by farzadha2** [reply]
Thanks for the reply, i assume it uses the click.rb
